00_download_data/nbconverted/2_download_cancer_gene_data.py

```diff
@@ -0,0 +1,290 @@
+
+# In[6]:
+
+
+import mpmp.utilities.data_utilities as du
+
+vogelstein_df = du.load_vogelstein()
+
+print(vogelstein_df.shape)
+vogelstein_df.head()
+
+
+# ### Clean up oncogene/TSG annotations and merge datasets
+# 
+# We want as many genes as possible to be annotated as _either_ an oncogene or TSG, so we know whether to use copy gain or copy loss data to define relevant CNV info. 
+# 
+# So, here, we will:
+# 
+# 1. Drop genes that are annotated only as fusion genes (since we're not calling fusions at this time)  
+# 2. Try to resolve genes that are annotated as both oncogene/TSG (usually context/cancer type specific) into their most likely pan-cancer category  
+# 3. For genes that can't be resolved confidently, we'll keep them as "oncogene, TSG" and run our scripts using both copy gain and copy loss downstream.
+# 4. Merge the three datasets together, ensuring that annotations are concordant (or resolving them if not)
+
+# In[7]:
+
+
+# first merge bailey and vogelstein datasets
+# these are both "confidently" annotated (everything is either an oncogene or TSG),
+# so we don't need to worry about resolving ambiguous annotations
+print(vogelstein_df.classification.unique())
+vogelstein_df.head()
+
+
+# In[8]:
+
+
+print(bailey_df.classification.unique())
+bailey_df.head()
+
+
+# In[9]:
+
+
+# first merge dataframes, then resolve classifications
+vogelstein_bailey_df = (
+    vogelstein_df.loc[:, ['gene', 'classification']]
+      .set_index('gene')
+      .merge(bailey_df.loc[:, ['classification']],
+             how='outer', left_index=True, right_index=True)
+      .rename(columns={'classification_x': 'vogelstein_classification',
+                       'classification_y': 'bailey_classification'})
+)
+
+print(vogelstein_bailey_df.shape)
+vogelstein_bailey_df.head()
+
+
+# In[10]:
+
+
+def merge_classifications(row):
+    if row['vogelstein_classification'] == row['bailey_classification']:
+        return row['vogelstein_classification']
+    elif pd.isna(row['vogelstein_classification']):
+        return row['bailey_classification']
+    elif pd.isna(row['bailey_classification']):
+        return row['vogelstein_classification']
+    elif row['vogelstein_classification'] != row['bailey_classification']:
+        # if the datasets disagree, we can resolve these manually
+        # or just run them as "oncogene, TSG"
+        return 'check'
+    else:
+        # not sure how this would happen
+        return pd.NA
+    
+vogelstein_bailey_df['classification'] = (
+    vogelstein_bailey_df.apply(merge_classifications, axis='columns')
+)
+
+print(vogelstein_bailey_df.classification.unique())
+vogelstein_bailey_df.head()
+
+
+# In[11]:
+
+
+# examples where the datasets disagree
+vogelstein_bailey_df[vogelstein_bailey_df.classification == 'check'].head()
```

Comment:
Very minor point, but I thought you were fixing only the first four discordant genes manually rather than all the discordant genes due to the `.head()` call here

Reply:
Ahh, got it. I guess I don't need the `head()` call since there's only four of them total.